editor/modes/visual-editor/block.js

```diff
@@ -276,78 +271,81 @@ class VisualEditorBlock extends wp.element.Component {
 	}
 }
 
-export default connect(
-	( state, ownProps ) => {
-		return {
-			previousBlock: getPreviousBlock( state, ownProps.uid ),
-			nextBlock: getNextBlock( state, ownProps.uid ),
-			block: getBlock( state, ownProps.uid ),
-			isSelected: isBlockSelected( state, ownProps.uid ),
-			isMultiSelected: isBlockMultiSelected( state, ownProps.uid ),
-			isFirstSelected: isFirstSelectedBlock( state, ownProps.uid ),
-			selectedBlocks: getSelectedBlocks( state ),
-			isHovered: isBlockHovered( state, ownProps.uid ),
-			focus: getBlockFocus( state, ownProps.uid ),
-			isTyping: isTypingInBlock( state, ownProps.uid ),
-			order: getBlockIndex( state, ownProps.uid ),
-		};
-	},
-	( dispatch, ownProps ) => ( {
-		onChange( uid, updates ) {
-			dispatch( {
-				type: 'UPDATE_BLOCK',
-				uid,
-				updates,
-			} );
-		},
-		onSelect() {
-			dispatch( {
-				type: 'TOGGLE_BLOCK_SELECTED',
-				selected: true,
-				uid: ownProps.uid,
-			} );
-		},
-		onDeselect() {
-			dispatch( { type: 'CLEAR_SELECTED_BLOCK' } );
-		},
-		onStartTyping() {
-			dispatch( {
-				type: 'START_TYPING',
-				uid: ownProps.uid,
-			} );
-		},
-		onHover() {
-			dispatch( {
-				type: 'TOGGLE_BLOCK_HOVERED',
-				hovered: true,
-				uid: ownProps.uid,
-			} );
-		},
-		onMouseLeave() {
-			dispatch( {
-				type: 'TOGGLE_BLOCK_HOVERED',
-				hovered: false,
-				uid: ownProps.uid,
-			} );
+export default flow(
+	applyComponentDecorators,
```

Comment:
This is exactly what we need to Collaborative Editing to be able to display a different UI state when collaborator is editing the block 👍